Endpoints for the typeScript server and .env file

etherpilled opened the pull request:
In this branch I created an endpoint (/trade?) and (/poolComposition) which can be used by GET() methods from any programming language to use DeFund sdk by the API. I also added multi-network capabilities to sign and to choose the infura rpc. The update also segragates the private keys and API keys into an .env.

Commit:
adding infura and private key in .env files

Commit:
Generalization

This commit generalizes the network that we can use for several functions and also facilitate the use of infura.io and secures the private keys and also the api keys in .env files.

Commit:
End points and Request Response

In this commit I added the trade endpoint that allows GET() requests to that endpoint with easier specification of the parameters for the trade_defund function and sdk usage. By allowing these type of calls we open the doors for multiple programming language to be able to operate with DeFund SDK through this new API.

Commit:
.env file for keys

Add here your private key and infura api key

Commit:
try/catch

adding try and catch for error handling.

Commit:
poolComposition

I added the pool Composition fund for the poolComposition end point.

Commit:
bump up defund sdk

mailbaoer commented on .env:
```diff
@@ -0,0 +1,2 @@
+PRIVATE_KEY=""
```
use .env.example is a better choice

etherpilled replied:
why ?

mailbaoer commented on README.md:
````diff
@@ -15,6 +15,13 @@ cd defund-express-examples
 npm install
 ```
 
+Create an .env file with two paramaters in the main folder (defund-express-example). (If you don't have an Infura project, set up one for free [here](https://infura.io/).)
````
There is no need to limit the rpc service provider, you can add several ways of writing, for example
1. new providers.AlchemyProvider(ChainId, AlchemyKey);
2. new providers.JsonRpcProvider('your rpc url');

etherpilled replied:
We can have infura by default, and have a "custom" for users to enter the full endpoint.

mailbaoer commented on src/defund.ts:
```diff
@@ -37,13 +37,35 @@ function networkToChainId(network: string): number {
 
-function rpcProvider() {
-    return new providers.JsonRpcProvider("your rpc url");
+function rpcProvider(network: string) {
```
you can just use `new providers.InfuraProvider(ChainId, InfuraKey);` for simple

etherpilled replied:
I will rewrite this for more customization

mailbaoer commented on src/index.ts:
```diff
@@ -1,32 +1,50 @@
+    const options = {
+            gasLimit: 1000000,
+            gasPrice: 60, // 60 gwei
+    }
```
maybe add optional comment is better

etherpilled replied:
what you mean ?

mailbaoer replied:
I mean that add a comment for options param is optional

etherpilled replied:
Ok

Commit:
Update README.md

Commit:
Update defund.ts

Commit:
better gas estimates

I added options for better polygon gas estimates

Commit:
Update defund.ts

Commit:
more options for RPC

now you can add a custom rpc provider and replace it into the getSigner function if desired.

Commit:
Update defund.ts
